### PR TITLE
[JENKINS-33309] Implement ChangeRequestAction

### PR DESCRIPTION
**Pull request by jglick**
Downstream of https://github.com/jenkinsci/scm-api-plugin/pull/7 and intended to supersede #28.

@reviewbybees esp. @imeredith

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/github_branch_source/PullRequestAction.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import hudson.model.InvisibleAction;
+import java.net.URL;
+import jenkins.scm.api.SCMHead;
+import jenkins.scm.api.actions.ChangeRequestAction;
+import org.kohsuke.github.GHPullRequest;
+
+/**
+ * Metadata about a {@link PullRequestSCMHead}.
+ */
+final class PullRequestAction extends ChangeRequestAction {
```
Comment (KostyaSha) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/PullRequestAction.java]: Smells like https://github.com/jenkinsci/github-integration-plugin/blob/master/github-pullrequest-plugin/src/main/java/org/jenkinsci/plugins/github/pullrequest/GitHubPRPullRequest.java#L23-L27

Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/PullRequestAction.java]: Yes, very similar.

**Comment by imeredith** [comment]
:bee: from me.

**Comment by recena** [comment]
:bee: Nice work @jglick

**Pull request merged**